econml/dml/causal_forest.py

```diff
@@ -498,7 +501,103 @@ def _gen_model_final(self):
     def _gen_rlearner_model_final(self):
         return _CausalForestFinalWrapper(self._gen_model_final(), False, self._gen_featurizer(), False)
 
+    def tune(self, Y, T, *, X=None, W=None,
+             sample_weight=None, sample_var=None, groups=None,
+             params='auto'):
+        """
+        Tunes the major hyperparameters of the final stage causal forest based on out-of-sample R-score
+        performance. It trains small forests of size 100 trees on a grid of parameters and tests the
+        out of sample R-score. After the function is called, then all parameters of `self` have been
+        set to the optimal hyperparameters found.

```

Comment:
now only final stage params are tunable

Reply:
I still think it's worth mentioning that this estimator has not been fit with this data, so that the user knows that the they still have to call fit afterward.

Reply:
Or alternatively, in your notebooks I think you always use:
```python3
est.tune(...)
est.fit(...)
```

Are there times a user would not want to do that, or should calling fit just be folded into tune so that it saves the user the trouble?

Reply:
That's what I'm doing in the notebooks I think. 

I think it's good to have separate. Maybe someone wants to tune on a subset of the data or  some small chunk. Also the tune does not need to take other keyword args like inference, cache_values etc.

Reply:
I added the extra docstring comment that the returned self is not yet fitted.